### ISSUE TITLE
Propagate POB change

### ISSUE THREAD

**Issue by Captainkirkdawson** [FreeUKGen/FreeCENMigration#934]
Here is a thought bubble:

 

How difficult is it to have a button and code such as

 

Select all from (file being validated)
Geoff requested function that:
Where verbatim_birth_county and verbatim_birth_place equals (Chapman code and POB of record being validated) AND birth_county and birth_place are null

Replace all  birth_county and birth_place with (birth_county and birth_place of record being validated)

The button would promulgate the entry through the piece. So every instance would have been validated.

It would be a huge time saver for validators as they only have to validate the entry once.

**Comment by Captainkirkdawson** [comment]
@geoffj-FUG can you confirm that the Alt county and place name need to be valid before propagation?

**Comment by geoffj-FUG** [comment]
Kirk

 

Yes please.

 

They must be valid. If not valid when the record is initially accessed then it must be added to the Place Names file either as a new Place or as an Alternative Place. 

 

I presume that the file would then be reprocessed and the Alt POB would test as valid and can be Promulgated through the piece. 
Or does #948 make this unnecessary?
 

Geoff

**Comment by Captainkirkdawson** [comment]
@geoffj-FUG; 948 simply confirms the change being added is valid; propagation could be an add on to this

**Comment by geoffj-FUG** [comment]
Kirk

 

I think that Propagate Alt POB should be on the screen in the second row after Accept Entry.

 

If the entry is not accepted then Propagation should not work.

 

I see it as 2 separate decisions by the validator: 

The first is that the entry with the alternative POB is valid and therefore accepted. 

The second is that the alternative POB should be pre-filled for every recurrence of the originally transcribed POB. 

It should only be prefilled, not pre-accepted. It will need validation but not re-typing. 

 

Geoff

**Comment by Captainkirkdawson** [comment]
@geoffj-FUG You may have noticed I have been reluctant to start this one. There is a reason. In REG we did a great deal of propagation; some of it was good and some was bad. It is the one thing that keeps coming back for me to fix when someone makes a mistake. Now there the situation is different as the propagation goes into the search records as well. In CEN since it is within validation that same issue will not occur. But once bitten twice shy. Will it be used enough to warrant the effort which I now estimate at several days.
The other question is Is this the next priority or are there others?. Is it required before release or can it be added later.
I will send a list of outstanding work.

**Comment by Captainkirkdawson** [comment]
@geoffj-FUG this has once again reached the top of the file. Before I hit the keyboard 

1. Can you please reconfirm that this is still a requirement

2. If it is can you confirm that you would like it to appear as a new option on the validation screen. With a small amount of work it could be made to only appear if the alternate fields were added.

3. That the propagation should be automatic but rather than an acceptance we add a warning that validation is required just as we do now if the alt fields are present.

**Comment by geoffj-FUG** [comment]
Kirk

 

1.	Yes this is still a requirement.

 

Have a look at RG13_2298 Warnings and you will see the problem. In this case the Transcriber has chosen to use the alternative place name fields. But not all Transcribers are so meticulous. If this had been left to the validator the validator would have had to repeatedly enter the alternative place name, just because the enumerator abbreviated ‘south’ every time.

 

Promulgation removes the repetitive task of retyping the alternative place of birth throughout the piece.

 

2.	Yes.
3.	Automatic? Let me bounce that off of John and Beki. It is definitely Warning not an automatic acceptance.

 

Geoff

**Comment by Captainkirkdawson** [comment]
@geoffj-FUG Will hold until 3 is answered.

**Comment by geoffj-FUG** [comment]
3. No. It should be initiated by the Validator. 
It will require the Validator to Re-Process the file to update the Warnings each time that it happens. If it was automatic that is a significant processing load when Promulgations are not needed. (Can this be done with the equivalent of what we used to call a batch file?).
The Re Processing will ensure that all Warnings are up to date.
I will write the instructions appropriately.
Geoff

**Comment by Captainkirkdawson** [comment]
@geoffj-FUG 
A first version of the Propagate action has been deployed to test3.
It works as follows. Make and edit to a record involving either or both of the Alternate POB fields. If the change is accepted ie the new fields are correct the display of the altered record displays a new action. Propagate Alternate Fields
![Propagate.png](https://images.zenhubusercontent.com/5b68ae684b5806bc2bdefcf0/a5342e80-52db-4e9c-9c13-ae48e8157cdc)
Selecting this action (which has the normal challenge) searches the file for other entries with the same verbatim fields (both must match) and updates the Alternate Fields for those records. A warning is added to them to say an adjustment was made.
![Propagate1.png](https://images.zenhubusercontent.com/5b68ae684b5806bc2bdefcf0/e703d717-f921-4750-9818-2252892fd833).

**Comment by geoffj-FUG** [comment]
Kirk
Brillaint!
I chose a piece that already had alternate place names in it and removed about 40 of them. I then changed the rest of the alternate place name entries from south to north.
I then set the piece to validate. I validated the first incorrect place name entry and the propogate button came up. I propogated and it filled in all the entries I had removed and left the ones that I had changed alone.
So in short, exactly what we need. It will save validators a lot of tedium.
Geoff
